src/utils/helpers.py

```diff
@@ -1,33 +1,73 @@
 import json
+import time
 from pathlib import Path
 
+import jwt
+import requests
 
-def load_predicates(path: str) -> list[str]:
+
+def create_jwt_token(gh_app_id: str, gh_pem_path: str) -> str:
     """
-    Fetch BETWEEN predicates from a provided SQL file.
+    Generates a JSON web token for authentication with the GitHub API.
 
     Args:
-        path: String path to a SQL file within the `config/`
-            directory. The SQL file should define SQL BETWEEN expressions,
-            where each expression is one chunk that will be extracted by
-            Spark during JDBC reads. Expressions should not be overlapping.
+        gh_app_id: The GitHub App ID.
+        gh_pem_path: The path to the GitHub App's private key file.
 
     Returns:
-        A list of SQL predicate strings used to divide a table into chunks
-        during JDBC reads.
+        str: The JWT token for authentication.
     """
-    full_path = "/tmp/config" / Path(path)
-    with open(full_path.resolve().as_posix(), mode="r") as file:
-        predicates = [line.strip() for line in file.readlines()]
 
-    return predicates
+    with open(gh_pem_path, "rb") as pem_file:
+        signing_key = jwt.jwk_from_pem(pem_file.read())
 
+    payload = {
+        "iat": int(time.time()),
+        "exp": int(time.time()) + 60,
+        "iss": gh_app_id,
+    }
 
-def strip_table_prefix(table_name: str) -> str:
-    """Removes the 'iasworld.' prefix from a table name."""
-    if table_name.startswith("iasworld."):
-        return table_name.split(".", 1)[1]
-    return table_name
+    jwt_instance = jwt.JWT()
+    encoded_jwt = jwt_instance.encode(payload, signing_key, alg="RS256")
+
+    return encoded_jwt
+
+
+def dispatch_gh_worfklow(gh_jwt: str, gh_repo: str, gh_workflow: str) -> None:
+    """
+    Dispatch a GitHub Actions workflow using the GitHub API.
+
+    Args:
+        gh_jwt: The JWT token for authentication with the GitHub API. Returned
+             by the `create_jwt_token` function.
+        gh_repo: API URL for the target repository containing a workflow.
+        gh_workflow: Workflow YAML file, relative to `.github/workflows`.
```

Comment:
**[Question, non-blocking]** Does this param require a file extension (`.yaml`)?

Reply:
It does! But this is obviated by the class-based refactor I just did.